sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructField.scala

```diff
@@ -74,4 +74,11 @@ case class StructField(
   def getComment(): Option[String] = {
     if (metadata.contains("comment")) Option(metadata.getString("comment")) else None
   }
+
+  /**
+   * Return the full description of the StructField.
+   */
+  def getDesc(): String = {
```

Comment:
`case class StructField` is external and marked as stable.

`getDesc ` is only used in the error message. We should avoid adding this.